jlibra-integration-tests/src/test/java/dev/jlibra/MoneyTransferIT.java

```diff
@@ -0,0 +1,200 @@
+import io.grpc.ManagedChannelBuilder;
+import org.apache.commons.lang3.RandomUtils;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.util.encoders.Hex;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.security.Security;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static admission_control.AdmissionControlOuterClass.AdmissionControlStatus.Accepted;
+import static dev.jlibra.mnemonic.Mnemonic.WORDS;
+import static java.lang.String.format;
+import static java.time.Instant.now;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.with;
+import static org.awaitility.pollinterval.FibonacciPollInterval.fibonacci;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * 1. Create two key pairs A and B.
+ * 2. Mint X libras for account represented by key pair A.
+ * 3. Transfer amount Y from A to B and verify the transaction.
+ */
+public class MoneyTransferIT {
+
+    private static final String TEST_SALT = "Salt, pepper and a dash of sugar.";
+    private static final String TESTNET_ADDRESS = "ac.testnet.libra.org";
+    private static final int TESTNET_PORT = 8000;
+
+    @BeforeClass
+    public static void setUpClass() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    private ManagedChannel channel;
+    private AdmissionControl admissionControl;
+    private ExtendedPrivKey sourceAccount;
+
+    @Before
+    public void setUp() {
+        // source account is fixed
+        Mnemonic sourceMnemonic = Mnemonic.fromString("hurry seven priority awful wear jeans antique add fetch sure negative finish suit draft myself chimney spend marine clock furnace draft public erase evidence");
+        Seed seed = new Seed(sourceMnemonic, "LIBRA");
+        LibraKeyFactory libraKeyFactory = new LibraKeyFactory(seed);
+        sourceAccount = libraKeyFactory.privateChild(new ChildNumber(0));
+
+        channel = ManagedChannelBuilder.forAddress(TESTNET_ADDRESS, TESTNET_PORT)
+                .usePlaintext()
+                .build();
+
+        admissionControl = new AdmissionControl(channel);
+    }
+
+    @After
+    public void tearDown() {
+        channel.shutdown();
+    }
+
+    @Test
+    public void transferTest() throws IOException {
+
+        // mint something to source account so we don't run out of coins
+        mint();
+
+        // destination account is generated
+        ExtendedPrivKey destination = generateKey();
+        String destinationAddress = destination.getAddress();
+        long transactionAmount = 1_000;
+
+        // make the transaction
+        transfer(destinationAddress, transactionAmount);
+
+        // wait for balance to become visible
```

Comment:
this awaitility I hadn't seen before, seems pretty cool and handy :thumbsup:

Reply:
good library for async http testing